packages/x-data-grid/src/hooks/features/sorting/gridSortingSelector.ts

```diff
@@ -25,8 +27,19 @@ export const gridSortedRowIdsSelector = createSelector(
 export const gridSortedRowEntriesSelector = createSelectorMemoized(
   gridSortedRowIdsSelector,
   gridRowsLookupSelector,
-  // TODO rows v6: Is this the best approach ?
-  (sortedIds, idRowsLookup) => sortedIds.map((id) => ({ id, model: idRowsLookup[id] ?? {} })),
+  gridRowTreeSelector,
+  (sortedIds, idRowsLookup, rowTree) =>
+    sortedIds.reduce<GridValidRowModel[]>((acc, id) => {
+      const model = idRowsLookup[id];
+      if (model) {
+        acc.push({ id, model });
+      }
+      const rowNode = rowTree[id];
+      if (isAutoGeneratedRow(rowNode)) {
```

Comment:
To align with
https://github.com/mui/mui-x/blob/ad0c9bfbf2da0497beccbb99aaae0c78de205ef4/packages/x-data-grid/src/hooks/features/rows/useGridRows.ts#L103-L106